borg/archive.py

```diff
@@ -263,8 +265,9 @@ def save(self, name=None, timestamp=None):
             'time': start.isoformat(),
             'time_end': end.isoformat(),
             'chunker_params': self.chunker_params,
-        })
-        data = msgpack.packb(metadata, unicode_errors='surrogateescape')
+        }
+        metadata.update(additional_metadata or {})
+        data = msgpack.packb(StableDict(metadata), unicode_errors='surrogateescape')
```

Comment:
This part likely merge-conflicts with #847. I'd say we merge #847 first and then I'll add "--comment" to this PR's "borg recreate" command, replacing the "borg comment" command from #847.